pywr/parameters/_parameters.pyx

```diff
@@ -1577,6 +1577,58 @@ cdef class NegativeMinParameter(MinParameter):
 NegativeMinParameter.register()
 
 
+cdef class OffsetParameter(Parameter):
+    """Parameter that takes offsets another `Parameter` by a constant value.
+
+    This class is a more efficient version of `AggregatedParameter` where
+    a single `Parameter` is offset by a constant value.
+
+    Parameters
+    ----------
+    parameter : `Parameter`
+        The parameter to to compare with the float.
+    offset : float (default=0.0)
+        The offset to add to apply to value returned by `parameter`.
+    lower_bounds : float (default=0.0)
+        The lower bounds of the offset when used during optimisation.
+    upper_bounds : float (default=np.inf)
+        The upper bounds of the offset when used during optimisation.
+    """
+    def __init__(self, model, parameter, offset=0.0, lower_bounds=0.0, upper_bounds=np.inf, *args, **kwargs):
+        super(OffsetParameter, self).__init__(model, *args, **kwargs)
+        self.parameter = parameter
+        self.children.add(parameter)
+        self.offset = offset
+        self.double_size = 1
+        self._lower_bounds = np.ones(self.double_size) * lower_bounds
+        self._upper_bounds = np.ones(self.double_size) * upper_bounds
```

Comment:
There are a few other parameters that need need to have these optimisation attributes setup in the same way (e.g. `MaxParameter`).

Could the parameters that have another parameter and a single float as input args all inherit from a single base class that defines these attributes and associated methods?

Reply:
I'm not sure how you would do this cleanly (e.g. this offset uses a different internal name). I think there could be a common base class for all of `MaxParameter`, `MinParameter`, `NegativeMaxParameter` and `NegativeMinParameter` to add this support. Probably one for another issue. It is also related to #632. What do you think?

Reply:
We should probably have a `MinMaxParameter` too.

Reply:
I agree it is a separate issue related to #632. Looks like there are quite a few parameters that people potentially might want to use as optimisation variables but are not currently setup with the new attributes.